File: guides/micronaut-token-propagation/userecho/kotlin/src/main/kotlin/example/micronaut/UserController.kt
package example.micronaut

import io.micronaut.http.MediaType.TEXT_PLAIN
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get
import io.micronaut.http.annotation.Produces
import io.micronaut.security.annotation.Secured
import io.micronaut.security.rules.SecurityRule.IS_AUTHENTICATED
import java.security.Principal

@Controller("/user") // <1>
class UserController {

    @Secured(IS_AUTHENTICATED) // <2>
    @Produces(TEXT_PLAIN) // <3>
    @Get // <4>
    fun index(principal: Principal): String = principal.name // <5>
}

File: guides/micronaut-token-propagation/intermediate-userecho/python/src/example/micronaut/user_controller.py
from java.security import Principal
from micronaut.http import MediaType
from micronaut.http.annotation import Get, Produces
from micronaut.security.annotation import Secured
from micronaut.security.rules import SecurityRule


# tag::clazz[]
@Secured(SecurityRule.IS_AUTHENTICATED)  # <2>
@Produces(MediaType.TEXT_PLAIN)  # <3>
@Get("/user")  # <1> <4>
def index(principal: Principal) -> str:  # <5>
    return principal.getName()
# end::clazz[]
